tests/attacks/test_backdoor_attack.py

```diff
@@ -0,0 +1,373 @@
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import unittest
+
+import numpy as np
+from keras import Sequential
+from keras.layers import Conv2D, MaxPooling2D, Dropout, Flatten, Dense
+
+from art.attacks.poisoning.backdoor_attack import PoisoningAttackBackdoor
+from art.attacks.poisoning.perturbations import add_pattern_bd, add_single_bd, insert_image
+from art.classifiers import KerasClassifier
+from art.utils import preprocess, load_mnist
+from tests.utils import TestBase, master_seed
+
+logger = logging.getLogger(__name__)
+
+PP_POISON = 0.33
+
+
+class TestBackdoorAttack(TestBase):
+    """
+    A unittest class for testing Backdoor Poisoning attack.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        master_seed(seed=1234)
+
+        (cls.x_train_mnist_raw, cls.y_train_mnist_raw), (cls.x_test_mnist_raw, cls.y_test_mnist_raw), \
+            cls._min, cls._max = load_mnist(raw=True)
+
+        cls.n_train_raw = 5000
+        cls.n_test_raw = 5000
+        cls.x_train_mnist_raw = cls.x_train_mnist_raw[0:cls.n_train_raw]
+        cls.y_train_mnist_raw = cls.y_train_mnist_raw[0:cls.n_train_raw]
+        cls.x_test_mnist_raw = cls.x_test_mnist_raw[0:cls.n_test_raw]
+        cls.y_test_mnist_raw = cls.y_test_mnist_raw[0:cls.n_test_raw]
+
+    def setUp(self):
+        master_seed(seed=1234)
+        super().setUp()
+
+    def get_keras_model(self):
+        model = Sequential()
+        model.add(Conv2D(32, kernel_size=(3, 3), activation='relu', input_shape=self.x_train_mnist.shape[1:]))
+        model.add(Conv2D(64, (3, 3), activation='relu'))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
+        model.add(Dropout(0.25))
+        model.add(Flatten())
+        model.add(Dense(128, activation='relu'))
+        model.add(Dropout(0.5))
+        model.add(Dense(10, activation='softmax'))
+
+        model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
+        classifier = KerasClassifier(model=model, clip_values=(self._min, self._max))
+        return classifier
```

Comment:
Would it be possible to use ART's standard testing classifier `art.tests.uitls.get_classifier_kr` instead?

Reply:
Fixed in commit